testing/citest/tests/kube_smoke_test.py

```diff
@@ -518,6 +669,18 @@ def test_a_create_app(self):
 
   def test_b_upsert_load_balancer(self):
     self.run_test_case(self.scenario.upsert_load_balancer())
+  def save_statefulset_pipeline(self):
```

Comment:
I'm pretty sure that tests have to start with 'test_' in order for the test runner to pick them up as a test case rather than a helper method. So I do not think these tests are actually being run.

Furthermore, there's a caveat in the test running in that it runs tests alphabetically rather than the order of declaration. That's something I never figured out how to fix easily and intended on changing anyway (though never got around to it).

That's why there's the test_a_... and so forth. So you need to sequence these. at least relative to each other if they have dependencies on one another or other tests. For example, you might need these to run between the test_a_create_app and test_z_delete_app. If so, you can treat a letter decorator as a phase, and add additional numbers or some other scheme after that. So for example:
test_c_1_create_statefulset_pipeline
test_c_2_...

You'll probably need to rename some of the existing tests. You can add a gap.
The create_server_group is independent of these so you can stick it in the same phase and it's order doesnt matter so name could remain the same, but htat might feel weird and unintentionally imply that it is part of that phase when really it doesnt matter. So at least a comment